### PR TITLE
Update Comment Removal

### PR DESCRIPTION
**Pull request by JPolacek**
###  Summary

The purpose of this PR is update the comment removal regex and add tests to ensure that the sanitizer handles comment removal in the manner that is desired per [this HTML documentation](https://html.spec.whatwg.org/multipage/parsing.html#parse-error-abrupt-closing-of-empty-comment).

_N.B. - you'll want to view this with whitespace changes turned off_
### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackhq/htmlsanitizer-hack/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've written tests to cover the new code and functionality included in this PR.

### CODE REVIEW

File: src/Lexer/DOMLex.hack [hunk 1]
```diff
@@ -9,227 +9,255 @@ use namespace HTMLPurifier\Token;
...
+	public function tokenizeHTML(
+		string $html,
+		HTMLPurifier\HTMLPurifier_Config $config,
+		HTMLPurifier\HTMLPurifier_Context $context,
+	): vec<HTMLPurifier\HTMLPurifier_Token> {
+		$html = $this->normalize($html, $config, $context);
+		// attempt to armor stray angled brackets that cannot possibly
+		// form tags and thus are probably being used as emoticons
+		if ($config->def->defaults['Core.AggressivelyFixLt']) {
+			while (Str\contains($html, '<!--')) {
```
Comment (id4ho) [inline src/Lexer/DOMLex.hack:32]: initially was confused by this cause the POC still renders in the browser if you add spaces (like `<! --><iframe srcdoc='<script>alert(1)</script>'>-->`) but it seems like later on we parse it in a way that drops the comments after this `while` cause I can't get the tests to break in the right way :thinking:

**Review by id4ho — COMMENTED** [review]
I think this looks pretty good. I'd like to review the regexes themselves a bit more next week.

Seems like this may be a good candidate for fuzzing 🤔

File: src/Lexer/DOMLex.hack [hunk 2]
```diff
@@ -9,227 +9,255 @@ use namespace HTMLPurifier\Token;
...
+		//     //need to clarify what we need to set libxml_pargehuge
+		//     $options = $options | \LIBXML_PARSEHUGE;
+		// }
+
+		\set_error_handler(meth_caller(HTMLPurifier_Lexer_DOMLex::class, 'muteErrorHandler'));
+		$doc->loadHTML($html, $options);
+		\restore_error_handler();
+		$body = $doc->getElementsByTagName('html')
+			->item(0)
+			-> //<html>
```
[the remark this answers is not in the document]
Comment (JPolacek) [reply src/Lexer/DOMLex.hack:73]: Yeah for sure - I think that was the autoformat but it's def weird

File: src/Lexer/DOMLex.hack [hunk 3]
```diff
@@ -9,227 +9,255 @@ use namespace HTMLPurifier\Token;
...
+		string $html,
+		HTMLPurifier\HTMLPurifier_Config $config,
+		HTMLPurifier\HTMLPurifier_Context $context,
+	): vec<HTMLPurifier\HTMLPurifier_Token> {
+		$html = $this->normalize($html, $config, $context);
+		// attempt to armor stray angled brackets that cannot possibly
+		// form tags and thus are probably being used as emoticons
+		if ($config->def->defaults['Core.AggressivelyFixLt']) {
+			while (Str\contains($html, '<!--')) {
+				$__unused_var = null;
```
[the remark this answers is not in the document]
Comment (JPolacek) [reply src/Lexer/DOMLex.hack:33]: Lol yeah, not sure how/why that's there ¯\_(ツ)_/¯

File: src/Lexer/DOMLex.hack (hunk 1 above)
Comment (JPolacek) [reply src/Lexer/DOMLex.hack:32]: Hmm interesting... We should look some more into this when we review the regexes

**Review by id4ho — APPROVED** [review]

**Pull request merged**